pkg/kademlia/kademlia.go

```diff
@@ -581,7 +581,16 @@ func (k *Kad) ClosestPeer(addr swarm.Address, skipPeers ...swarm.Address) (swarm
 
 	// check if self
 	if closest.Equal(k.base) {
-		return swarm.Address{}, topology.ErrWantSelf
+		switch l := len(skipPeers); l {
```

Comment:
on second  thought this is not  a good idea. 
also disconnected peers etc clearly shows this should be an iterator function.
either way, checking  the length of skipPeers is a hack and not the right thing.
For instance if we had to prepopulate skipPeers with some nodes, which seems legit, suddenly, 
the closest node will not be found. 

Instead you need to check closest before you  check skippeers and  simply never return errwantself iff there is a closer peer even if it is to be skipped.